### PR TITLE
[SECURESIGN-844] Migrate to kubebuilder go/v4

### PR DESCRIPTION
**Pull request by bouskaJ**

### CODE REVIEW

**Comment by bouskaJ** [comment]
in order to pass `tas-operator-e2e` https://github.com/openshift/release/pull/52925 needs to be merged

File: Makefile [hunk 1]
```diff
@@ -117,52 +123,69 @@ vet: ## Run go vet against code.
...
 # More info: https://docs.docker.com/develop/develop-images/build_enhancements/
 .PHONY: docker-build
-docker-build: test ## Build docker image with the manager.
-	docker build . -t ${IMG}
+docker-build: test## Build docker image with the manager.
+	$(CONTAINER_TOOL) build -t ${IMG} .
 
 .PHONY: docker-build-skip-test
 docker-build-skip-test: ## Build docker image with the manager.
 	docker build . -t ${IMG}
```
Comment (osmman) [inline Makefile:168]: it should use `$(CONTAINER_TOOL)`

File: Makefile [hunk 2]
```diff
@@ -196,23 +219,23 @@ $(LOCALBIN):
...
+KUBECTL ?= kubectl
 KUSTOMIZE ?= $(LOCALBIN)/kustomize
 CONTROLLER_GEN ?= $(LOCALBIN)/controller-gen
 ENVTEST ?= $(LOCALBIN)/setup-envtest
 
 ## Tool Versions
-KUSTOMIZE_VERSION ?= v3.8.7
-CONTROLLER_TOOLS_VERSION ?= v0.14.0
+KUSTOMIZE_VERSION ?= v5.2.1
+CONTROLLER_TOOLS_VERSION ?= v0.13.0
```
Comment (osmman) [inline Makefile:229]: We should use original version and do not downgrade.

File: Makefile
Comment (osmman) [inline Makefile:1]: https://github.com/securesign/secure-sign-operator/pull/431/files#diff-76ed074a9305c04054cdebb9e9aad2d818052b07091de1f20cad0bbac34ffb52L269 should be changed to use `$(CONTAINER_TOOL)`

File: bundle/manifests/rhtas.redhat.com_ctlogs.yaml
Comment (osmman) [inline bundle/manifests/rhtas.redhat.com_ctlogs.yaml:1]: please regenerate all manifests files with newer version 0.14.0 of controller-gen

File: bundle.Dockerfile [hunk 3]
```diff
@@ -6,7 +6,7 @@ LABEL operators.operatorframework.io.bundle.manifests.v1=manifests/
 LABEL operators.operatorframework.io.bundle.metadata.v1=metadata/
 LABEL operators.operatorframework.io.bundle.package.v1=rhtas-operator
 LABEL operators.operatorframework.io.bundle.channels.v1=stable,stable-v1.0
-LABEL operators.operatorframework.io.metrics.builder=operator-sdk-v1.32.0
+LABEL operators.operatorframework.io.metrics.builder=operator-sdk-v1.34.0
```
Comment (osmman) [inline bundle.Dockerfile:9]: There are used multiple version of operator sdk in the project. It should be fixed to use single one.
I saw 1.34.0, 1.34.1, 1.34.2

File: cmd/main.go [hunk 4]
```diff
@@ -86,25 +88,28 @@ func init() {
...
-		pprofAddr            string
 		enableLeaderElection bool
+		probeAddr            string
+		secureMetrics        bool
+		enableHTTP2          bool
 	)
+
 	flag.StringVar(&metricsAddr, "metrics-bind-address", ":8080", "The address the metric endpoint binds to.")
 	flag.StringVar(&probeAddr, "health-probe-bind-address", ":8081", "The address the probe endpoint binds to.")
-	flag.StringVar(&pprofAddr, "pprof-address", "", "The address to expose the pprof server. Default is empty string which disables the pprof server.")
```
Comment (osmman) [inline cmd/main.go:104]: We need this one, it has been added by me.

File: cmd/main.go [hunk 5]
```diff
@@ -130,14 +135,35 @@ func main() {
...
 		Metrics: metricsserver.Options{
-			BindAddress: metricsAddr,
+			BindAddress:   metricsAddr,
+			SecureServing: secureMetrics,
+			TLSOpts:       tlsOpts,
 		},
-		// Port:                   9443,
+		WebhookServer:          webhookServer,
 		HealthProbeBindAddress: probeAddr,
-		PprofBindAddress:       pprofAddr,
```
Comment (osmman) [inline cmd/main.go:140]: return it back

**Review by osmman — CHANGES_REQUESTED** [review]

**Comment by osmman** [comment]
`.tekton` pipelines need to be updated to use new paths. Controller and main.go has been moved so it should test internal, api and cmd folders

https://github.com/securesign/secure-sign-operator/blob/06fade2d547bc5f5d5c8d0a67d4bf19f512ccc23/.tekton/rhtas-operator-push.yaml#L10
https://github.com/securesign/secure-sign-operator/blob/06fade2d547bc5f5d5c8d0a67d4bf19f512ccc23/.tekton/rhtas-operator-pull-request.yaml#L11

File: Makefile (hunk 2 above)
Comment (osmman) [reply Makefile:229]: `make generate` command is failing on null pointer exception with 0.13.0. It will pass when I upgrade it to 0.14.0

**Review by osmman — CHANGES_REQUESTED** [review]
Please regenerate config and bundle resources by controller-gen v0.14.0

**Comment by bouskaJ** [comment]
@bouskaJ File paths changed. Can you check all other components for `build.appstudio.openshift.io/build-nudge-files: "internal/controller/constants/*"`?

**Pull request merged**